hail/python/hailtop/batch/backend.py

```diff
@@ -184,18 +190,17 @@ def run_code(code):
         copied_input_resource_files = set()
         os.makedirs(tmpdir + '/inputs/', exist_ok=True)
 
-        if batch.requester_pays_project:
-            requester_pays_project = f'-u {batch.requester_pays_project}'
-        else:
-            requester_pays_project = ''
+        requester_pays_project_json = orjson.dumps(batch.requester_pays_project).decode('utf-8')
 
         def copy_input(job, r):
             if isinstance(r, resource.InputResourceFile):
                 if r not in copied_input_resource_files:
                     copied_input_resource_files.add(r)
 
                     if r._input_path.startswith('gs://'):
-                        return [f'gsutil {requester_pays_project} cp -r {shq(r._input_path)} {shq(r._get_path(tmpdir))}']
+                        transfers_bytes = orjson.dumps([{"from": r._input_path, "to": r._get_path(tmpdir)}])
```

Comment:
Do we still need the `if r._input_path.startswith('gs://'):` here or should this work everywhere now?

Reply:
bump

Reply:
It looks like, in the local backend, we sometimes use a symlink instead of a copy, so I tried to preserve that behavior. Admittedly, I didn't change the local-to-local-outside-docker to use copy tool, it just uses cp still.

Reply:
ok. I think we'll want some global variable for the available schemes and check if the input path starts with one of the valid schemes.

Reply:
Ah, sorry, I misunderstood your comment. Hmm. Yeah, you're right this should support hail-az, s3, and gs.

Reply:
OK, here's a fix: https://github.com/hail-is/hail/pull/11091/commits/878177ab4363ddd5466790dce9d56d91d8af798b. If it has no scheme e.g. `/foo/bar`, then we presume its local. Otherwise, even `file:/`, we use the copy tool. That makes sense because `cp` doesn't understand `file:/` anyway.